python/lsst/jointcal/testUtils.py

```diff
@@ -0,0 +1,290 @@
+
+    Parameters
+    ----------
+    num1, num2 : `int`, optional
+        Number of sources to put in the first and second catalogs. Should be
+        a square, to have sqrt(num) centroids on a grid.
+    seed : `int`, optional
+        Seed value for np.random.
+
+    Returns
+    -------
+    struct : `lsst.pipe.base.Struct`
+       Result struct with components:
+
+       - `camera` : Camera representing these catalogs
+           (`lsst.afw.cameraGeom.Camera`).
+       - `catalogs` : Catalogs containing fake sources
+           (`list` of `lsst.afw.table.SourceCatalog`).
+       - `ccdImageList` : CcdImages containing the metadata and fake sources
+           (`list` of `lsst.jointcal.CcdImage`).
+       - `bbox` : Bounding Box of the image (`lsst.afw.geom.Box2I`).
+    """
+    np.random.seed(seed)
+
+    visit1 = 849375
+    visit2 = 850587
+    ccdId = 12
+    instFluxKeyName = "SomeFlux"
+
+    # Load or fake the necessary metadata for each CcdImage
+    dataDir = lsst.utils.getPackageDir('jointcal')
+    inputDir = os.path.join(dataDir, 'tests/data/cfht_minimal')
+    butler = lsst.daf.persistence.Butler(inputDir)
+
+    # so we can access parts of the camera later (e.g. focal plane)
+    camera = butler.get('camera', visit=visit1, ccd=ccdId)
+
+    struct1 = createFakeCcdImage(butler, visit1, ccdId, num1, instFluxKeyName,
+                                 photoCalibMean=100.0, photoCalibErr=1.0)
+    struct2 = createFakeCcdImage(butler, visit2, ccdId, num2, instFluxKeyName,
+                                 photoCalibMean=120.0, photoCalibErr=5.0)
+
+    return lsst.pipe.base.Struct(camera=camera,
+                                 catalogs=[struct1.catalog, struct2.catalog],
+                                 ccdImageList=[struct1.ccdImage, struct2.ccdImage],
+                                 bbox=struct1.bbox)
+
+
+def createFakeCcdImage(butler, visit, ccdId, num, instFluxKeyName,
+                       photoCalibMean=100.0, photoCalibErr=1.0):
+    """Create a fake CcdImage by making a fake catalog.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.persistence.Butler`
+        Butler to load metadata from.
+    visit : `int`
+        Visit identifier to build a butler dataId.
+    ccdId : `int`
+        CCD identifier to build a butler dataId.
+    num : `int`
+        Number of sources to put in the catalogs. Should be
+        a square, to have sqrt(num) centroids on a grid.
+    instFluxKeyName : `str`
+        Name of the instFluxKey to populate in the catalog.
+    photoCalibMean : `float`, optional
+        Value to set for calibrationMean in the created PhotoCalib.
+    photoCalibErr : `float`, optional
+        Value to set for calibrationErr in the created PhotoCalib.
+
+    Returns
+    -------
+    struct : `lsst.pipe.base.Struct`
+       Result struct with components:
+
+       - `catalog` : Catalogs containing fake sources
+           (`lsst.afw.table.SourceCatalog`).
```

Comment:
Please swap the type and the description here and for `ccdImage` immediately below and clean up `bbox` immediately after that. I guess it's not a `Parameters` list and maybe it doesn't matter, but it's a bit odd being different than parameter lists for no readily apparently reason (and different item by item).

Reply:
That's how Struct returns are supposed to be documented: https://developer.lsst.io/python/numpydoc.html#struct-return-types

Reply:
Oh, and the reason `bbox` looks different is because it's on the same line.